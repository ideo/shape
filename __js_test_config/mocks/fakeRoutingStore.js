const fakeRoutingStore = {
  previousPageBeforeSearch: null,
  pathTo: jest.fn(),
  routeTo: jest.fn(),
  pathContains: jest.fn(),
  updatePreviousPageBeforeSearch: jest.fn(),
  leaveSearch: jest.fn(),
}

export default fakeRoutingStore
