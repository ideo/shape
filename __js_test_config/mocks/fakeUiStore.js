const fakeUiStore = {
  gridSettings: {
    cols: 4,
    gutter: 20,
    gridW: 312,
    gridH: 250,
  },
  blankContentToolState: {
    order: null,
    width: null,
    height: null,
    replacingId: null,
  },
  dialogConfig: {
    open: null,
    prompt: null,
    onConfirm: null,
    onCancel: null,
    iconName: null,
    confirmText: 'OK',
    cancelText: 'Cancel',
    onClose: jest.fn(),
  },
  activityLogOpen: false,
  activityLogPosition: {
    x: 0,
    y: 0,
    h: 1,
    w: 1,
  },
  activityLogMoving: false,
  scroll: {
    scrollToTop: jest.fn(),
    scrollToBottom: jest.fn(),
  },
  cardMenuOpen: {
    id: false,
    x: 0,
    y: 0,
  },
  closeBlankContentTool: jest.fn(),
  closeCardMenu: jest.fn(),
  openCardMenu: jest.fn(),
  closeMoveMenu: jest.fn(),
  resetSelectionAndBCT: jest.fn(),
  rolesMenuOpen: false,
  isLoading: false,
  selectedCardIds: [],
  selectCardId: jest.fn(),
  setViewingCollection: jest.fn(),
  setViewingItem: jest.fn(),
  viewingCollection: null,
  movingFromCollectionId: null,
  movingCardIds: [],
  openMoveMenu: jest.fn(),
  update: jest.fn(),
  alert: jest.fn(),
  alertOk: jest.fn(),
  defaultAlertError: jest.fn(),
  confirm: jest.fn(),
  closeDialog: jest.fn(),
  cardAction: 'move',
  blurContent: false,
  organizationMenuPage: 'organizationMenuPage',
  organizationMenuGroupId: null,
  expandedThreadKey: null,
  expandThread: jest.fn(),
  openGroup: jest.fn(),
  openOptionalMenus: jest.fn(),
  trackEvent: jest.fn(),
  trackedRecords: {},
  editingName: false,
  activityLogPage: 'comments',
  pageMenuOpen: false,
  searchText: '',
  collectionCardSortOrder: '',
  addNewCard: jest.fn(),
  removeNewCard: jest.fn(),
  isNewCard: jest.fn(),
  editingCardId: 0,
  toggleEditingCardId: jest.fn(),
  autocompleteMenuClosed: jest.fn(),
}

export default fakeUiStore
