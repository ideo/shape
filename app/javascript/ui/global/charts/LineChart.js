import PropTypes from 'prop-types'
import { VictoryLine } from 'victory'

import ChartTooltip from '~/ui/global/charts/ChartTooltip'
import {
  datasetPropType,
  dateTooltipText,
  advancedTooltipText,
  addDuplicateValueIfSingleValue,
  chartDomainForDatasetValues,
  lineChartDashWithForOrder,
} from '~/ui/global/charts/ChartUtils'

const formatValues = values => {
  const formatted = addDuplicateValueIfSingleValue(values)
  return formatted.map((datum, i) => ({
    ...datum,
    x: i + 1,
    y: datum.value,
  }))
}

const chartStyle = dataset => {
  const { style } = dataset
  return {
    data: {
      stroke: (style && style.fill) || '#000000',
      strokeWidth: 3,
      strokeDasharray: lineChartDashWithForOrder({
        order: dataset.order,
        scale: 1.5,
      }),
    },
  }
}

const LineChart = ({ dataset, simpleDateTooltip, cardArea }) => {
  const { measure, timeframe } = dataset
  const values = formatValues(dataset.data)
  const domain = chartDomainForDatasetValues({
    values,
    maxDomain: dataset.max_domain,
  })
  let tooltipFn
  if (simpleDateTooltip) {
    tooltipFn = datum => dateTooltipText(datum)
  } else {
    tooltipFn = (datum, isLastDataPoint) =>
      advancedTooltipText({
        datum,
        isLastDataPoint,
        timeframe,
        measure,
      })
  }
  return (
    <VictoryLine
      labels={d => d.value}
      labelComponent={
        <ChartTooltip textRenderer={tooltipFn} cardArea={cardArea} />
      }
      style={chartStyle(dataset)}
      data={values}
      domain={domain}
      key={`dataset-${dataset.measure}`}
    />
  )
}

LineChart.propTypes = {
  dataset: datasetPropType.isRequired,
  simpleDateTooltip: PropTypes.bool,
  cardArea: PropTypes.number,
}

LineChart.defaultProps = {
  cardArea: 1,
  simpleDateTooltip: false,
}

export default LineChart
