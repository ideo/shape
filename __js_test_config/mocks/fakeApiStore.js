import fakeUiStore from './fakeUiStore'

const fakeApiStore = ({
  findResult = '',
  findAllResult = [],
  requestResult = { data: {} },
} = {}) => {
  return {
    add: jest.fn().mockName('add'),
    removeAll: jest.fn().mockName('removeAll'),
    find: jest.fn().mockReturnValue(findResult),
    findAll: jest.fn().mockReturnValue(findAllResult),
    request: jest
      .fn()
      .mockReturnValue(Promise.resolve(requestResult))
      .mockName('request'),
    fetch: jest
      .fn()
      .mockReturnValue(Promise.resolve(requestResult))
      .mockName('fetch'),
    fetchRoles: jest
      .fn()
      .mockReturnValue(Promise.resolve())
      .mockName('fetchRoles'),
    sync: jest.fn().mockName('sync'),
    loadCurrentUser: jest.fn().mockReturnValue(Promise.resolve()),
    searchUsersAndGroups: jest
      .fn()
      .mockReturnValue(Promise.resolve({ data: [] })),
    searchUsers: jest
      .fn()
      .mockReturnValue(Promise.resolve({ data: [] })),
    searchCollections: jest.fn().mockReturnValue(Promise.resolve({ data: [] })),
    loadCurrentUserGroups: jest.fn(),
    searchRoles: jest.fn().mockReturnValue(Promise.resolve({ data: [] })),
    setCurrentUserId: jest.fn(),
    findOrganizationById: jest.fn().mockReturnValue({ name: 'abc' }),
    fetchThreads: jest.fn().mockReturnValue(Promise.resolve()),
    fetchNotifications: jest.fn().mockReturnValue(Promise.resolve()),
    findOrBuildCommentThread: jest
      .fn()
      .mockReturnValue(Promise.resolve({ key: 'x' })),
    findThreadForRecord: jest.fn(),
    loadNextThreadPage: jest.fn(),
    createTemplateInstance: jest.fn(),
    moveCards: jest.fn(),
    linkCards: jest.fn(),
    duplicateCards: jest.fn(),
    checkInMyCollection: jest.fn(),
    unreadActivityCount: 0,

    usersThreadPagesToLoad: 1,
    loadingThreads: false,
    hasOlderThreads: false,

    currentUserId: '1',
    currentUser: {
      id: '1',
      current_user_collection_id: 99,
      current_organization: {
        name: 'Acme Inc',
        deactivated: false,
      },
      organizations: [{ name: 'Acme Inc 1', filestack_file_url: 'test.jpg' }],
      name: 'Johnny Appleseed',
      pic_url_square: 'https://s3.amazonaws.com/pic.png',
      groups: [],
      terms_accepted: true,
      switchOrganization: jest.fn(),
      API_hideHelper: jest.fn(),
    },
    // NOTE: important that this matches currentUserOrganization.slug
    currentOrgSlug: 'org-slug',
    currentUserOrganization: {
      id: '1',
      slug: 'org-slug',
      name: 'Acme Inc',
      primary_group: {
        name: 'Acme Inc',
      },
    },
    unreadNotifications: [],
    recentNotifications: new Map(),
    currentThreads: [
      {
        id: '1',
        key: 'abc-1',
        record: { name: 'abc' },
        comments: [],
        users_thread: {
          subscribed: true,
        },
      },
      {
        id: '2',
        key: 'abc-2',
        record: { name: 'xyz' },
        comments: [],
        users_thread: {
          subscribed: true,
        },
      },
    ],
    collections: [],
    items: [],

    uiStore: fakeUiStore,
    undoStore: {},
    routingStore: {},
  }
}

export default fakeApiStore
