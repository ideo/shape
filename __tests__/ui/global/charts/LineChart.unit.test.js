import LineChart from '~/ui/global/charts/LineChart'
import { fakeAreaChartDataset } from '#/mocks/data'

const props = {}
let wrapper, render

describe('LineChart', () => {
  beforeEach(() => {
    props.dataset = fakeAreaChartDataset
    props.simpleDateTooltip = true
    render = () => (wrapper = shallow(<LineChart {...props} />))
    render()
  })

  it('renders VictoryLine', () => {
    expect(wrapper.find('VictoryLine').exists()).toBe(true)
  })
})
