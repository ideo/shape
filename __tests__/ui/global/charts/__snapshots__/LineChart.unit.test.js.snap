// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart renders snapshot 1`] = `
<VictoryLine
  containerComponent={
    <VictoryContainer
      className="VictoryContainer"
      portalComponent={<Portal />}
      portalZIndex={99}
      responsive={true}
    />
  }
  data={
    Array [
      Object {
        "date": "2018-07-10",
        "value": 10,
        "x": 1,
        "y": 10,
      },
      Object {
        "date": "2018-08-10",
        "value": 25,
        "x": 2,
        "y": 25,
      },
      Object {
        "date": "2018-09-10",
        "value": 30,
        "x": 3,
        "y": 30,
      },
    ]
  }
  dataComponent={
    <Curve
      pathComponent={<Path />}
    />
  }
  domain={
    Object {
      "x": Array [
        1,
        3,
      ],
      "y": Array [
        0,
        30,
      ],
    }
  }
  groupComponent={
    <VictoryClipContainer
      circleComponent={<Circle />}
      clipPathComponent={<ClipPath />}
      groupComponent={<g />}
      rectComponent={<Rect />}
    />
  }
  key="dataset-0"
  labelComponent={
    <ChartTooltip
      alwaysShowLabels={false}
      cardArea={1}
      displayTicks={true}
      labelTextRenderer={[Function]}
      maxValue={0}
      minValue={0}
      tooltipTextRenderer={[Function]}
    />
  }
  labels={[Function]}
  samples={50}
  scale="linear"
  sortKey="x"
  sortOrder="ascending"
  standalone={true}
  style={
    Object {
      "data": Object {
        "stroke": "#120f0e",
        "strokeDasharray": "3,6",
        "strokeWidth": 3,
      },
    }
  }
  theme={
    Object {
      "area": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#252525",
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "axis": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "axis": Object {
            "fill": "transparent",
            "stroke": "#252525",
            "strokeLinecap": "round",
            "strokeLinejoin": "round",
            "strokeWidth": 1,
          },
          "axisLabel": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 25,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
          "grid": Object {
            "fill": "none",
            "pointerEvents": "painted",
            "stroke": "none",
          },
          "tickLabels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "ticks": Object {
            "fill": "transparent",
            "size": 1,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "bar": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#252525",
            "padding": 8,
            "strokeWidth": 0,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "boxplot": Object {
        "boxWidth": 20,
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "max": Object {
            "padding": 8,
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "maxLabels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "median": Object {
            "padding": 8,
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "medianLabels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "min": Object {
            "padding": 8,
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "minLabels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "q1": Object {
            "fill": "#969696",
            "padding": 8,
          },
          "q1Labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "q3": Object {
            "fill": "#969696",
            "padding": 8,
          },
          "q3Labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
        },
        "width": 450,
      },
      "candlestick": Object {
        "candleColors": Object {
          "negative": "#252525",
          "positive": "#ffffff",
        },
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "chart": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "width": 450,
      },
      "errorbar": Object {
        "borderWidth": 8,
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "#252525",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "group": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "width": 450,
      },
      "legend": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "gutter": 10,
        "orientation": "vertical",
        "style": Object {
          "data": Object {
            "type": "circle",
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
          },
          "title": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 5,
            "stroke": "transparent",
          },
        },
        "titleOrientation": "top",
      },
      "line": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "#252525",
            "strokeWidth": 2,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "pie": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 400,
        "padding": 50,
        "style": Object {
          "data": Object {
            "padding": 10,
            "stroke": "transparent",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 20,
            "stroke": "transparent",
          },
        },
        "width": 400,
      },
      "scatter": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "#252525",
            "stroke": "transparent",
            "strokeWidth": 0,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 10,
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
      "stack": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "width": 450,
      },
      "tooltip": Object {
        "cornerRadius": 5,
        "flyoutStyle": Object {
          "fill": "#f0f0f0",
          "pointerEvents": "none",
          "stroke": "#252525",
          "strokeWidth": 1,
        },
        "pointerLength": 10,
        "style": Object {
          "fill": "#252525",
          "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
          "fontSize": 14,
          "letterSpacing": "normal",
          "padding": 5,
          "pointerEvents": "none",
          "stroke": "transparent",
          "textAnchor": "middle",
        },
      },
      "voronoi": Object {
        "colorScale": Array [
          "#252525",
          "#525252",
          "#737373",
          "#969696",
          "#bdbdbd",
          "#d9d9d9",
          "#f0f0f0",
        ],
        "height": 300,
        "padding": 50,
        "style": Object {
          "data": Object {
            "fill": "transparent",
            "stroke": "transparent",
            "strokeWidth": 0,
          },
          "flyout": Object {
            "fill": "#f0f0f0",
            "pointerEvents": "none",
            "stroke": "#252525",
            "strokeWidth": 1,
          },
          "labels": Object {
            "fill": "#252525",
            "fontFamily": "'Gill Sans', 'Gill Sans MT', 'Seravek', 'Trebuchet MS', sans-serif",
            "fontSize": 14,
            "letterSpacing": "normal",
            "padding": 5,
            "pointerEvents": "none",
            "stroke": "transparent",
            "textAnchor": "middle",
          },
        },
        "width": 450,
      },
    }
  }
/>
`;
